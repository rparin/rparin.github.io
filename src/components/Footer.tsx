import Image from "next/image";
import extLink from "@icons/external-link.svg";

export default function Footer() {
  return (
    <footer className="text-center text-white bg-black py-3 md:py-5">
      <div className="flex flex-row text-center justify-center">
        <p>Built using React and Tailwind.</p>
        <a
          href="https://github.com/rparin/rparin.github.io"
          target="_blank"
          rel="noopener noreferrer">
          <Image
            className="max-w-[1.5vh] invert w-auto h-3/5"
            src={extLink}
            alt="External link to source code"
          />
        </a>
      </div>
      <p>©️ 2023. Designed by Ralph Parin.</p>
    </footer>
  );
}
