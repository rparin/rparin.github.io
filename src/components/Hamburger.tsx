const hamburgerStyle = {
  width: "1rem",
  display: "flex",
  justifyContent: "spaceâ€”around",
  flexFlow: "column nowrap",
  zIndex: 10,
  gap: ".25em",
};

const burgerStyle = {
  width: "2rem",
  height: "0.25rem",
  borderRadius: "10rem",
  backgroundColor: "white",
};

export default function Hamburger() {
  var burgers = [];
  for (var i = 0; i < 3; i++) {
    burgers.push(<div style={burgerStyle} key={i}></div>);
  }

  return <button style={hamburgerStyle}>{burgers}</button>;
}
