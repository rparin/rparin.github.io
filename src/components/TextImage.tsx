import React from "react";
import { cn } from "@/lib/utils";
import { cva, type VariantProps } from "class-variance-authority";
import Image, { StaticImageData } from "next/image";

const TextImageVariants = cva(
  "absolute bg-background rounded-lg line-clamp-2 md:line-clamp-none top-1/4 md:top-1/3 p-2 md:p-3 text-sm md:text-base md:max-w-lg lg:max-w-xl",
  {
    variants: {
      variant: {
        left: "left-[20%] md:left-[30%] lg:left-[35%]",
        right: "right-[25%] lg:right-[33%]",
      },
    },
    defaultVariants: {
      variant: "left",
    },
  }
);

export interface TextImageProps
  extends React.ButtonHTMLAttributes<HTMLButtonElement>,
    VariantProps<typeof TextImageVariants> {
  text: string;
  img: StaticImageData;
  alt: string;
}

const TextImage = React.forwardRef<HTMLButtonElement, TextImageProps>(
  ({ className, variant, text, img, alt }, ref) => {
    return (
      <figure
        ref={ref}
        className={cn("relative flex justify-center", className)}>
        <figcaption className={cn(TextImageVariants({ variant }))}>
          {text}
        </figcaption>
        <Image
          className="max-h-48 md:max-h-56 lg:max-h-64 w-auto"
          src={img}
          alt={alt}
        />
      </figure>
    );
  }
);

export { TextImage, TextImageVariants };
