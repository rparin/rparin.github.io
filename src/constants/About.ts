const AboutGen = {
  Initials: "RP",
  First: "Ralph",
  Last: "Parin",
  email: "r.parin070@gmail.com",
  titles: "Software Engineer, Web Developer, Gamer",
  sent: "I enjoy",
};

const AboutSent = [
  "reading light novels",
  "playing games",
  "watching movies",
  "learning new things",
];

const Socials = {
  github: "https://github.com/rparin",
  linkedIn: "https://www.linkedin.com/in/rparin",
};

const AboutMe = {
  Intro: "A little about me: my hobbies, interests, and goals.",

  Goals:
    "The thrill of crafting with my hands led me to computer science, where I aspire to make a meaningful difference in the lives of others, whether through a useful digital tool or a fun game.",
  GoalsSm:
    "Crafting drove me to computer science to make a difference, via games or digital tools.",

  Hobbies:
    "I cherish stories for the unique lessons they offer, particularly the no-happy-ending tales by my favorite author, Sugaru Miaki. They've taught me to embrace failure as a path to improvement.",
  HobbiesSm:
    "I enjoy reading for the lessons they offer. They've, taught me to overcome my regrets.",

  Interests:
    "One of my passion lies in creating memorable moments with friendsâ€”whether it's trying new flavors at local eateries, embarking on food adventures, or exploring exciting new places together.",
  InterestsSm:
    "One hobby of mine includes trying new foods and sightseeing with friends.",

  Gaming:
    "Gaming with my friends has become my digital backyard, as life takes us on different paths and distances grow, this space keeps us connected.",
  GamingSm:
    "Gaming with friends is my digital backyard, keeping me close despite life's changes.",
};

export { AboutGen, AboutSent, Socials, AboutMe };
