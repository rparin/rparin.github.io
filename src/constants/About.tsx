const AboutGen = {
  First: "Ralph",
  Last: "Parin",
  email: "r.parin070@gmail.com",
  titles: "Software Engineer, WebDeveloper, Gamer",
  sent: "I enjoy",
};

const AboutSent = [
  "reading light novels.",
  "playing games.",
  "watching movies.",
  "trying new things.",
];

export { AboutGen, AboutSent };
