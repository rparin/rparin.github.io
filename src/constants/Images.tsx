import landBgR from "../assets/Landing/Red.svg";
import landBgP from "../assets/Landing/Purple.svg";
import landCode from "../assets/Landing/CodePy.webp";
import AboutCode from "../assets/About/CodeE.webp";
import AboutFood from "../assets/About/FoodE.webp";
import AboutGame from "../assets/About/LaptopE.webp";
import AboutBook from "../assets/About/BookE.webp";

const HomeImages = {
  bgR: {
    img: landBgR,
    alt: "Landing page background",
  },
  bgP: {
    img: landBgP,
    alt: "Landing page background",
  },
  code: {
    img: landCode,
    alt: "Landing page code background overlay",
  },
};

const AboutImages = {
  code: {
    img: AboutCode,
    alt: "Background image of study room",
  },
  book: {
    img: AboutBook,
    alt: "Background image of book in forest",
  },
  food: {
    img: AboutFood,
    alt: "Background image of food in cafe",
  },
  game: {
    img: AboutGame,
    alt: "Background image of laptop in urban city",
  },
};

export { HomeImages, AboutImages };
