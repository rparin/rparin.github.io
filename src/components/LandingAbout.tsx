import { ThemeToggle } from "@/components/ui/theme-toggle";
import Image from "next/image";
import { AboutGen, AboutSent, Socials } from "@/constants/About";
import { Icons } from "@/constants/Images";

export default function LandingAbout() {
  return (
    <>
      <a
        href="#home"
        className="select-none flex gap-1 text-white md:text-black dark:text-white text-center text-lg md:text-xl absolute md:fixed top-2 left-5 md:left-10 z-50 px-2 hover:text-sky-400 active:text-accent dark:active:text-sky-400">
        <span className="text-base">&lt; </span>
        {AboutGen.Initials}
        <span className="text-base"> /&gt;</span>
      </a>

      <ThemeToggle className="absolute md:fixed top-1 right-5 md:right-10 z-50 border-0 h-9 rounded-md px-3 md:h-11 md:px-8 bg-background hover:bg-background active:bg-accent dark:bg-background dark:hover:bg-background dark:active:bg-accent" />

      <div className="absolute bottom-4 z-50 text-center w-full flex flex-col gap-1 md:gap-3">
        <h1 className="text-3xl md:text-4xl leading-10 md:leading-[3rem] font-semibold">
          {AboutGen.First} {AboutGen.Last}
        </h1>
        <h2 className="text-lg md:text-xl font-normal dark:font-light">
          {AboutGen.titles}
        </h2>
        <p className="text-lg md:text-xl font-normal dark:font-light">
          {AboutGen.sent}
          <span className="font-normal dark:font-light"> {AboutSent[3]}</span>
        </p>

        <div className="flex flex-row justify-center gap-3">
          <a
            href={"mailto:" + AboutGen.email}
            className="text-lg md:text-xl font-normal dark:font-light text-sky-500">
            {AboutGen.email}
          </a>
          <a href={Socials.github} target="_blank" rel="noopener noreferrer">
            <Image
              className="h-8 w-8 md:h-10 md:w-10 object-cover dark:invert"
              src={Icons.github.img}
              alt={Icons.github.alt}
            />
          </a>
          <a href={Socials.linkedIn} target="_blank" rel="noopener noreferrer">
            <Image
              className="h-8 w-8 md:h-10 md:w-10 object-cover"
              src={Icons.linkedin.img}
              alt={Icons.linkedin.alt}
            />
          </a>
        </div>
      </div>
    </>
  );
}
