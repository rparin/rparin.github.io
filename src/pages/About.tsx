import { AboutMe } from "@/constants/About";
import { AboutImages } from "@/constants/Images";
import { TextImage } from "@/components/TextImage";

export default function About({ id }: { id: string }) {
  return (
    <section
      id={id}
      className="mx-5 before:block before:h-20 before:content-[''] lg:mx-16">
      <div className="flex flex-col gap-10">
        <h2 className="text-center text-base font-normal md:text-lg">
          <i>{AboutMe.Intro}</i>
        </h2>

        <TextImage
          variant={"left"}
          text={AboutMe.Goals}
          textSm={AboutMe.GoalsSm}
          img={AboutImages.code.img}
          alt={AboutImages.code.alt}
        />
        <TextImage
          variant={"right"}
          text={AboutMe.Hobbies}
          textSm={AboutMe.HobbiesSm}
          img={AboutImages.book.img}
          alt={AboutImages.book.alt}
        />
        <TextImage
          variant={"left"}
          text={AboutMe.Interests}
          textSm={AboutMe.InterestsSm}
          img={AboutImages.food.img}
          alt={AboutImages.food.alt}
        />
        <TextImage
          variant={"right"}
          text={AboutMe.Gaming}
          textSm={AboutMe.GamingSm}
          img={AboutImages.game.img}
          alt={AboutImages.game.alt}
        />
      </div>
    </section>
  );
}
