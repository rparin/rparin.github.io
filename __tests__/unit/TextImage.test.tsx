import { render } from "@testing-library/react";
import { axe, toHaveNoViolations } from "jest-axe";
import { TextImage } from "@/components/TextImage";
import { AboutImages } from "@/constants/Images";
import { AboutMe } from "@/constants/About";

const PROPS = {
  text: AboutMe.Gaming,
  img: AboutImages.game.img,
  alt: AboutImages.game.alt,
};

describe("testing TextImage Component Accessibility", () => {
  expect.extend(toHaveNoViolations);
  test("it should have no accessibility violations", async () => {
    const { container } = render(<TextImage {...PROPS} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

test("it renders component unchanged", () => {
  const { container } = render(<TextImage {...PROPS} />);
  expect(container).toMatchSnapshot();
});
