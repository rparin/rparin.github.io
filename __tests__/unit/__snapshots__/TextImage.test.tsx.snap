// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component unchanged 1`] = `
<div>
  <figure
    class="relative flex justify-center"
  >
    <figcaption
      class="absolute bg-background rounded-lg line-clamp-2 md:line-clamp-none top-1/4 md:top-1/3 p-2 md:p-3 text-sm md:text-base md:max-w-lg left-[25%] md:left-[30%] lg:left-[38%]"
    >
      <span
        class="hidden md:block"
      >
        Gaming with my friends has become my digital backyard, as life takes us on different paths and distances grow, this space keeps us connected.
      </span>
      <span
        class="md:hidden"
      />
    </figcaption>
    <img
      alt="Background image of laptop in urban city"
      class="max-h-48 w-auto md:max-h-56 lg:max-h-64"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </figure>
</div>
`;
